Sources/AppCoinsSDK/UI/Purchase/PurchaseStateViews/CreditCardChoiceBottomSheet.swift

```diff
@@ -29,111 +30,280 @@ internal struct CreditCardChoiceBottomSheet: View {
     
     internal var body: some View {
         
-        VStack(spacing: 0) {
-            
-            if let storedPaymentMethods = adyenController.session?.sessionContext.paymentMethods.stored {
-                
-                Text(Constants.chooseCard)
-                    .font(FontsUi.APC_Body_Bold)
-                    .foregroundColor(ColorsUi.APC_Black)
-                    .frame(width: UIScreen.main.bounds.width - 64, height: 22, alignment: .leading)
-                    .padding(.top, 24)
-                    .padding(.bottom, 18)
-                
-                VStack(spacing: 0) {
-                    ForEach(0 ..< storedPaymentMethods.count, id: \.self) { paymentMethodNumber in
-                        
-                        if let paymentMethod = storedPaymentMethods[paymentMethodNumber] as? StoredCardPaymentMethod {
-                            Button(action: { self.chosenStoredCard = paymentMethod }) {
-                                ZStack {
-                                    ColorsUi.APC_White
-                                    HStack(spacing: 0) {
+        if let storedPaymentMethods = adyenController.session?.sessionContext.paymentMethods.stored {
+            if storedPaymentMethods.count > 2 {
```

Comment:
Can we find an easier solution for this? What guarantees us that 2 stored credit cards is a treshold from which we will need to have a scroll? This can vary by device and we should find a solution that is maintainable if possible.